src/chain/mod.rs

```diff
@@ -207,66 +314,41 @@ impl ChainSource {
 	) {
 		match self {
 			Self::Esplora { sync_config, logger, .. } => {
-				// Setup syncing intervals if enabled
-				if let Some(background_sync_config) = sync_config.background_sync_config {
-					let onchain_wallet_sync_interval_secs = background_sync_config
-						.onchain_wallet_sync_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut onchain_wallet_sync_interval = tokio::time::interval(
-						Duration::from_secs(onchain_wallet_sync_interval_secs),
-					);
-					onchain_wallet_sync_interval
-						.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
-
-					let fee_rate_cache_update_interval_secs = background_sync_config
-						.fee_rate_cache_update_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut fee_rate_update_interval = tokio::time::interval(Duration::from_secs(
-						fee_rate_cache_update_interval_secs,
-					));
-					// When starting up, we just blocked on updating, so skip the first tick.
-					fee_rate_update_interval.reset();
-					fee_rate_update_interval
-						.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
-
-					let lightning_wallet_sync_interval_secs = background_sync_config
-						.lightning_wallet_sync_interval_secs
-						.max(WALLET_SYNC_INTERVAL_MINIMUM_SECS);
-					let mut lightning_wallet_sync_interval = tokio::time::interval(
-						Duration::from_secs(lightning_wallet_sync_interval_secs),
+				if let Some(background_sync_config) = sync_config.background_sync_config.as_ref() {
```

Comment:
This extract could have been an independent commit. Non-blocking.

Reply:
I mean the extraction of code into `start_tx_based_sync_loop`. In a prefactor commit, this could be done for just Esplora. And then later electrum is reusing it.

Reply:
Still fine